### PR TITLE
Changes requested on PR970

### PR DESCRIPTION
**Pull request by adriacidre**
Some last minute changes couldn't be delivered as part of https://github.com/status-im/status-go/pull/970

Important changes:
- [x] Implementation wasn't successfully limiting requests 
- [x] Simplified body and test cases of TestMailServer

### CODE REVIEW

**Comment by adriacidre** [comment]
@adambabik this PR fixes changes on PR-970 if everything looks right, we can just merge this PR instead ofhttps://github.com/status-im/status-go/pull/980.

File: mailserver/mailserver_test.go [hunk 1]
```diff
@@ -163,129 +162,108 @@ func (s *MailserverSuite) TestDBKey() {
 	s.Equal(byte(i/0x1000000), k.raw[0], "big endian expected")
 }
 
-func (s *MailserverSuite) TestMailServer() {
-	var server WMailServer
+func (s *MailserverSuite) TestMailServerResponse() {
```
Comment (adambabik) [inline mailserver/mailserver_test.go]: We can remove this, right?

File: mailserver/mailserver_test.go [hunk 2]
```diff
@@ -163,129 +162,108 @@ func (s *MailserverSuite) TestDBKey() {
...
+			params: func() *ServerTestParams {
+				params := s.defaultServerParams(env)
+				params.upp = params.birth + 1
+				params.topic[0] = 0xFF
+
+				return params
+			}(),
+			expect: false,
+			isOK:   true,
+			info:   "Processing a request where to is grat than any existing register and with a specific topic, should not provide results",
```
Comment (adambabik) [inline mailserver/mailserver_test.go]: Typo: `grat`.

**Review by adambabik — APPROVED** [review]

File: mailserver/mailserver_test.go (hunk 1 above)
Comment (themue) [inline mailserver/mailserver_test.go]: ???

File: mailserver/mailserver_test.go (hunk 2 above)
Comment (themue) [inline mailserver/mailserver_test.go]: s/grat/greater/

File: mailserver/mailserver_test.go [hunk 3]
```diff
@@ -163,129 +162,108 @@ func (s *MailserverSuite) TestDBKey() {
...
+			params: func() *ServerTestParams {
+				params := s.defaultServerParams(env)
+				params.low = params.birth + 1
+				params.upp = params.birth + 1
+
+				return params
+			}(),
+			expect: false,
+			isOK:   true,
+			info:   "Processing a request where from and to are great than any existing register, should not provide results",
```
Comment (themue) [inline mailserver/mailserver_test.go]: s/great/greater/

File: mailserver/mailserver_test.go [hunk 4]
```diff
@@ -163,129 +162,108 @@ func (s *MailserverSuite) TestDBKey() {
...
-			params:      s.defaultServerParams(env),
-			lowModifier: 0,
-			uppModifier: 0,
-			expect:      true,
-			shouldFail:  false,
-			info:        "Processing a request where from and to are equals to an existing register, should provide results",
+			params: s.defaultServerParams(env),
+			expect: true,
+			isOK:   true,
+			info:   "Processing a request where from and to are equals to an existing register, should provide results",
```
Comment (themue) [inline mailserver/mailserver_test.go]: s/equals/equal/

**Review by themue — APPROVED** [review]
LGTM, only few remarks to infos.

**Pull request merged**